src/server/scripts/Pet/pet_dk.cpp

```diff
@@ -74,9 +68,28 @@ class npc_pet_dk_ebon_gargoyle : public CreatureScript
                     owner->RemoveAurasDueToSpell(SPELL_DK_SUMMON_GARGOYLE_2);
             }
 
+            void IsInteruptSpell(SpellInfo const* spell)
+            {
+                if (spell->HasEffect(SPELL_EFFECT_INTERRUPT_CAST))
+                {
+                    interrupted = true;
+                    events.ScheduleEvent(EVENT_INTERRUPT_OVER, 20s);
+                }
+            }
+
+            bool CheckCast()
+            {
+                if (me->IsWithinMeleeRange(me->GetVictim()))
+                    return roll_chance_i(40);
```

Comment:
???

Reply:
According to the logic of Peta needs to choose any target, after the killing of a mob.